integration/test/performancetest/performance_test.go

```diff
@@ -29,8 +35,24 @@ func TestPerformance(t *testing.T) {
 	test.StartAgent(configOutputPath, true)
 
 	agentRunDuration := agentRuntimeMinutes * time.Minute
-	//let agent run before collecting performance metrics on it
-	time.Sleep(agentRunDuration)
+
+	//create wait group so main test thread waits for log writing to finish before stopping agent and collecting data
+	var logWaitGroup sync.WaitGroup
+	logWaitGroup.Add(2)
+
+	//start goroutines to write to log files concurrently
+	go func() {
+		defer logWaitGroup.Done()
+		writeToLogs(t, logOutputPath1, agentRunDuration)
+	}()
+	go func() {
+		defer logWaitGroup.Done()
+		writeToLogs(t, logOutputPath2, agentRunDuration)
+	}()
+
+	//wait until writing to logs finishes
+	logWaitGroup.Wait()
```

Comment:
I would pull all of this out of the main function since we don't really need the wait group anywhere else other than to control synchronous writing to the log files for the duration of the test